atc/syslog/drainer.go

```diff
@@ -49,51 +50,66 @@ func (d *drainer) Run(ctx context.Context) error {
 			logger.Error("Syslog drainer connecting to server error.", err)
 			return err
 		}
-		defer syslog.Close()
+
+		// ignore any errors coming from syslog.Close()
+		defer db.Close(syslog)
 
 		for _, build := range builds {
-			events, err := build.Events(0)
+			err := d.drainBuild(logger, build, syslog)
 			if err != nil {
-				logger.Error("Syslog drainer getting build events error.", err)
 				return err
 			}
+		}
+	}
+	return nil
+}
 
-			for {
-				ev, err := events.Next()
-				if err != nil {
-					if err == db.ErrEndOfBuildEventStream {
-						break
-					}
-					logger.Error("Syslog drainer getting next event error.", err)
-					return err
-				}
-
-				if ev.Event == "log" {
-					var log event.Log
-
-					err := json.Unmarshal(*ev.Data, &log)
-					if err != nil {
-						logger.Error("Syslog drainer unmarshalling log error.", err)
-						return err
-					}
-
-					payload := log.Payload
-					tag := build.TeamName() + "/" + build.PipelineName() + "/" + build.JobName() + "/" + build.Name() + "/" + string(log.Origin.ID)
-
-					err = syslog.Write(d.hostname, tag, time.Unix(log.Time, 0), payload)
-					if err != nil {
-						logger.Error("Syslog drainer sending to server error.", err)
-						return err
-					}
-				}
+func (d *drainer) drainBuild(logger lager.Logger, build db.Build, syslog *Syslog) error {
+	events, err := build.Events(0)
+	if err != nil {
+		logger.Error("Syslog drainer getting build events error.", err)
+		return err
+	}
+
+	// ignore any errors coming from events.Close()
+	defer db.Close(events)
+
+	for {
+		ev, err := events.Next()
+		if err != nil {
+			if err == db.ErrEndOfBuildEventStream {
+				break
 			}
+			logger.Error("Syslog drainer getting next event error.", err)
```

Comment:
Making use of sessions, there's no need to prefix every log line with `Syslog drainer` given that we already have the context prefixed in the message (each `task` specified in a `session` gets appended with dots in the `message` field.

For instance, consider the following example where we have two actions being logged (`an-action` and `a-failed-action`) under the same session (as they share the same execution context):

```go
package main

import (
	"fmt"
	"os"

	"code.cloudfoundry.org/lager"
)

func main() {
	logger := lager.NewLogger("my-app")
	logger.RegisterSink(lager.NewWriterSink(os.Stdout, lager.DEBUG))

	sess := logger.Session("my-session")

	sess.Info("an-action")
	sess.Error("a-failed-action", fmt.Errorf("total failure"))
}
```

producing

```
{"timestamp":"1549320221.283070803","source":"my-app","message":"my-app.my-session.an-action","log_level":1,"data":{"session":"1"}}
{"timestamp":"1549320221.283279657","source":"my-app","message":"my-app.my-session.a-failed-action","log_level":2,"data":{"error":"total failure","session":"1"}}
```

Another detail: those actions don't have white spaces separating the fields, but instead, dashes - a pattern adopted across the whole codebase. 

See lager's [`Logger` interface](https://github.com/cloudfoundry/lager/blob/54c4f2530ddeb742ce9c9ee26f5fa758d6c1b3b6/logger.go#L16-L19) for more details on that.

Please let me know if you'd need more context!

Reply:
All of the logging styles were from before so I do think this is a good time to unify it with the rest of the code base

Reply:
Yeah we should probably put logging style somewhere in our wiki.

Reply:
@topherbullock got us covered! https://github.com/concourse/concourse/wiki/Concourse-Go-Style-Guide#logging